profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,153 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the redis database"
+  type        = string
+}
+
+variable "mongodb_version" {
+  description = "The version of mongodb. If null, the current default ICD mongodb version is used."
+  type        = string
+  default     = null
+}
+
+variable "plan" {
+  type        = string
+  description = "The name of the service plan that you choose for your mongodb instance"
+  default     = "standard"
+}
+
+variable "region" {
+  description = "The region mongodb is to be created on. The region must support BYOK ( us-south, us-east, and eu-de)"
+  type        = string
+  default     = "us-south"
+  validation {
+    condition = anytrue([
+      var.region == "us-south",
+      var.region == "us-east",
+      var.region == "eu-de"
+    ])
+    error_message = "region must be in a BYOK location, us-south, us-east or eu-de"
```

Comment:
I am not sure this is good, having just been performing a similar check for cross regional COS buckets. I think the high availability KP locations are us-south, eu-de and jp-tok. See https://cloud.ibm.com/docs/key-protect?topic=key-protect-ha-dr#availability. BYOK may be a different list.
see: https://ibm-cloudplatform.slack.com/archives/C01CSELQ2G2/p1677679833859209?thread_ts=1677151584.693849&cid=C01CSELQ2G2

Reply:
I wonder if durable BYOK from us-south or eu-de will be required to protect persistent data. SC-28(3) does not specify, so I expect any BYOK is sufficient.